### PR TITLE
add supported np.trapezoid

### PR DESCRIPTION
**Pull request by holymonson**
numpy 2.0 rename np.trapz to np.trapezoid

Ref: 
https://numpy.org/doc/stable/numpy_2_0_migration_guide.html#main-namespace
https://numpy.org/doc/stable/reference/generated/numpy.trapezoid.html
https://github.com/numpy/numpy/blob/b3ddf2fd33232b8939f48c7c68a61c10257cd0c5/numpy/lib/_function_base_impl.py#L5054-L5069

### CODE REVIEW

**Comment by kc611** [comment]
Hey @holymonson :wave: 

Thanks for contirbuting to Numba. The changes look good. :+1: 

Could you add a small release note into this PR so that your change can be displayed in the changelog of the next release. [Here's how to do that](https://github.com/numba/numba/tree/main/docs/upcoming_changes#readme)

Additionally, Could you add a test case, (which can be the same as the current `trapz` test) so that we'd be able to detect if the behaviour of the new `trapezoid` function  changes in NumPy in it's further releases. [Here's where `trapz` function is being tested](https://github.com/numba/numba/blob/a344e8f55440c91d40c5221e93a38ce0c149b803/numba/tests/test_np_functions.py#L3898)

File: docs/upcoming_changes/9719.np_support.rst [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Support for np.trapezoid
+------------------------
+
+Add Support for numpy 2.0 new function ``numpy.trapezoid``.
```
Comment (guilhermeleobas) [inline docs/upcoming_changes/9719.np_support.rst:4]:
```suggestion
Add support for numpy 2.0 new function ``numpy.trapezoid``.
```

File: numba/np/new_arraymath.py [hunk 2]
```diff
@@ -2236,6 +2236,11 @@ def impl(y, x=None, dx=1.0):
     return impl
 
 
+# numpy 2.0 rename np.trapz to np.trapezoid
+if numpy_version >= (2, 0):
+    overload(np.trapezoid)(np_trapz)
```
Comment (guilhermeleobas) [inline numba/np/new_arraymath.py:2241]: Does `np.trapz` still works in NumPy 2.0+?

Comment (holymonson) [reply numba/np/new_arraymath.py:2241]: Yes, with a warning.

```python
@set_module('numpy')
def trapz(y, x=None, dx=1.0, axis=-1):
    """
    `trapz` is deprecated in NumPy 2.0.

    Please use `trapezoid` instead, or one of the numerical integration
    functions in `scipy.integrate`.
    """
    # Deprecated in NumPy 2.0, 2023-08-18
    warnings.warn(
        "`trapz` is deprecated. Use `trapezoid` instead, or one of the "
        "numerical integration functions in `scipy.integrate`.",
        DeprecationWarning,
        stacklevel=2
    )
    return trapezoid(y, x=x, dx=dx, axis=axis)
```

**Comment by holymonson** [comment]
CI test matrix hasn't include numpy 2.0 yet.

**Review by guilhermeleobas — APPROVED** [review]

**Review by kc611 — APPROVED** [review]

**Comment by guilhermeleobas** [comment]
Thanks for your contribution @holymonson

**Comment by esc** [comment]
Thank you for the patch.

**Pull request merged**